noir/aztec_macros/src/lib.rs

```diff
@@ -199,6 +199,11 @@ fn lambda(parameters: Vec<(Pattern, UnresolvedType)>, body: Expression) -> Expre
 }
 
 macro_rules! chained_path {
+    ( $base:expr ) => {
```

Comment:
If there are no tail expressions we don't need to iterate over them, therefore mutable not needed and warning fixed